packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -870,7 +870,7 @@ export interface ISharedObjectKind<TSharedObject> {
  * @sealed
  * @public
  */
-export interface SharedObjectKind<out TSharedObject = unknown>
+export interface SharedObjectKind<out TSharedObject extends IFluidLoadable = IFluidLoadable>
```

Comment:
I think there are 3 aspects which we should think about for this specific line change:

1. When a type does not inherently require a particular type constraint, but its (current) usages do, is it better to have the constraint on the original type or not? This is a subjective style question. Personally I prefer not including unnecessary constraints just because some usages of the type require them, Those usages can have the constraints to avoid duplication of them and allot the types use to be less restricted. I specifically made that choice here. You can add the constraint to createSharedObjectKind if you want without adding it here: thats a separate choice (though adding the constraint here removes that choice from all users). Even if you want to keep the rest of this PR, I think omitting this line from it would make sense.
2. Does this prevent any usage we might want to allow/use in the future? Making this change makes it impossible to do some possibly useful upcasts. Suppose someone wanted to upcast a SharedObjectKind<Tree | Map> to a SharedObjectKind<IEventProvider<IErrorEvent>>. Such code used to work, and is a class of thing we might want to do when type erasing various parts of the APi in some cases, and your change breaks it.
3. Is this change actually allowed from a breaking changes perspective? GIven my example above, this is a breaking change. I suspect it won't break real apps, but it could so we would have to make it with care and a warning.

Given this, I'd recommend against this specific change. That does not means the rest of the PR shouldn't happen: thats separate (I believe)

Reply:
Naturally this is the one line that matters. Everything else is fallout from this change.